### PR TITLE
Hypervisor Manager support for API 800,1000,1200

### PR DESCRIPTION
**Pull request by nabhajit-ray**
Hypervisor Manager support for API 800,1000,1200

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,13 @@
 All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html)
 
+# [v1.3.0]
```
Comment (AnushaY1916) [inline CHANGELOG.md:4]: Make it as unreleased.

File: docs/d/hypervisor_manager.html.markdown [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+}
+```
+
+## Argument Reference
+
+* `uri` - (Required) The name of the hypervisor manager.
+
+## Attributes Reference
+
+* `fabric_type` - The supported Fibre Channel access method.
```
Comment (AnushaY1916) [inline docs/d/hypervisor_manager.html.markdown:31]: The attributes specified here belong to some other resource, add the attributes and its respective description of hypervisor manager resource.

File: examples/hypervisor_manager.tf [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	ov_sslverify = false
+	ov_apiversion = 1200
+	ov_ifmatch = "*"
+}
+
+data "oneview_scope" "scope_obj" {
+        name = "test_scope"
+}
+/*
+resource "oneview_hypervisor_manager" "HypervisorManager" {
```
Comment (AnushaY1916) [inline examples/hypervisor_manager.tf:14]: Comment either update or create requests, but not both, and write the description for these requests like "uncomment the following for update".

File: oneview/data_source_hypervisor_manager.go [hunk 4]
```diff
@@ -0,0 +1,245 @@
+// (C) Copyright 2016 Hewlett Packard Enterprise Development LP
```
Comment (AnushaY1916) [inline oneview/data_source_hypervisor_manager.go:1]: year change needed.

File: oneview/data_source_hypervisor_manager.go [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+		Read: dataSourceHypervisorManagerRead,
+
+		Schema: map[string]*schema.Schema{
+			"available_dvs_versions": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+			},
+
```
Comment (AnushaY1916) [inline oneview/data_source_hypervisor_manager.go:29]: Remove extra empty lines in schema

File: oneview/resource_hypervisor_manager.go [hunk 6]
```diff
@@ -0,0 +1,395 @@
...
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"etag": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"hypervisor_type": {
```
Comment (AnushaY1916) [inline oneview/resource_hypervisor_manager.go:64]: As per REST API documentation, hypervisorType should be optional not Computed.

File: oneview/resource_hypervisor_manager.go [hunk 7]
```diff
@@ -0,0 +1,395 @@
...
+	for _, raw := range resourcePathList {
+		resourcePaths := raw.(map[string]interface{})
+		hypervisorManagerResourcePaths := ov.ResourcePath{
+			UserPath:   resourcePaths["user_path"].(string),
+			ActualPath: resourcePaths["actual_path"].(string),
+		}
+		resourcePathCollect = append(resourcePathCollect, hypervisorManagerResourcePaths)
+		hypMan.ResourcePaths = resourcePathCollect
+	}
+	file, _ := json.MarshalIndent(hypMan, "", "")
```
Comment (AnushaY1916) [inline oneview/resource_hypervisor_manager.go:260]: Remove these lines as this is for debugging purposes.

**Comment by nabhajit-ray** [comment]
> Did you test importing an existing resource from Oneview?

Yes, the import testcases got passed

File: oneview/resource_hypervisor_manager.go (hunk 7 above)
Comment (nabhajit-ray) [reply oneview/resource_hypervisor_manager.go]: DOne

File: oneview/resource_hypervisor_manager.go (hunk 6 above)
Comment (nabhajit-ray) [reply oneview/resource_hypervisor_manager.go:57]: Done. Changed it to Optional

File: oneview/data_source_hypervisor_manager.go (hunk 5 above)
Comment (nabhajit-ray) [reply oneview/data_source_hypervisor_manager.go]: Done. Removed.

File: oneview/data_source_hypervisor_manager.go (hunk 4 above)
Comment (nabhajit-ray) [reply oneview/data_source_hypervisor_manager.go]: Done

File: examples/hypervisor_manager.tf (hunk 3 above)
Comment (nabhajit-ray) [reply examples/hypervisor_manager.tf:14]: Done

File: docs/d/hypervisor_manager.html.markdown (hunk 2 above)
Comment (nabhajit-ray) [reply docs/d/hypervisor_manager.html.markdown]: Done

File: CHANGELOG.md (hunk 1 above)
Comment (nabhajit-ray) [reply CHANGELOG.md:4]: Done

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,6 +1,13 @@
 All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html)
 
+# [v1.3.0](unreleased)
+### Notes
+- This release supports for Hypervisor Manager for API 800,1000,1200
```
Comment (jyothis-gireesan) [inline CHANGELOG.md:6]: You dont need to specify the resource name here as well as below. Just mention it as API * for the resources listed below.

File: oneview/provider.go [hunk 9]
```diff
@@ -96,6 +97,7 @@ func Provider() terraform.ResourceProvider {
 			"oneview_server_profile":             resourceServerProfile(),
 			"oneview_enclosure":                  resourceEnclosure(),
 			"oneview_enclosure_group":            resourceEnclosureGroup(),
+			"oneview_hypervisor_manager":         resourceHypervisorManager(),
```
Comment (jyothis-gireesan) [inline oneview/provider.go:100]: Make this in alphabetical order.

File: docs/d/hypervisor_manager.html.markdown [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+output "oneview_hypervisor_manager_value" {
+ value = "${data.oneview_hypervisor_manager.test.uri}"
+}
+```
+
+## Argument Reference
+
+* `uri` - (Required) The name of the hypervisor manager.
```
Comment (AnushaY1916) [inline docs/d/hypervisor_manager.html.markdown:27]: The attribute and its description are ambiguous. It should be name actually.

File: oneview/provider.go [hunk 11]
```diff
@@ -68,49 +68,52 @@ func Provider() terraform.ResourceProvider {
...
+			"oneview_logical_enclosure":          resourceLogicalEnclosure(),
 			"oneview_logical_interconnect_group": resourceLogicalInterconnectGroup(),
 			"oneview_logical_interconnect":       resourceLogicalInterconnect(),
 			"oneview_logical_switch_group":       resourceLogicalSwitchGroup(),
-			"oneview_uplink_set":                 resourceUplinkSet(),
-			"oneview_i3s_plan":                   resourceI3SPlan(),
-			"oneview_logical_enclosure":          resourceLogicalEnclosure(),
+			"oneview_network_set":                resourceNetworkSet(),
+			"oneview_scope":                      resourceScope(),
+			"oneview_server_certificate":         resourceServerCertificate(),
```
Comment (AnushaY1916) [inline oneview/provider.go:110]: ServerCertificate is not supported yet in this PR. Remove this entry.

Comment (nabhajit-ray) [reply oneview/provider.go]: Done

File: oneview/provider.go (hunk 9 above)
Comment (nabhajit-ray) [reply oneview/provider.go]: Done

File: CHANGELOG.md (hunk 8 above)
Comment (nabhajit-ray) [reply CHANGELOG.md]: Done

File: examples/hypervisor_manager.tf [hunk 12]
```diff
@@ -0,0 +1,48 @@
+provider "oneview" {
```
Comment (jyothis-gireesan) [inline examples/hypervisor_manager.tf:1]: Change actual values for placeholders.

File: oneview/data_source_hypervisor_manager_test.go [hunk 13]
```diff
@@ -0,0 +1,40 @@
+// (C) Copyright 2019 Hewlett Packard Enterprise Development LP
```
Comment (jyothis-gireesan) [inline oneview/data_source_hypervisor_manager_test.go:1]: Change the copyright year to 2020 for all the files.

File: oneview/resource_hypervisor_manager_test.go [hunk 14]
```diff
@@ -0,0 +1,98 @@
+// (C) Copyright 2016 Hewlett Packard Enterprise Development LP
```
Comment (jyothis-gireesan) [inline oneview/resource_hypervisor_manager_test.go:1]: Change the copyright to 2020

File: examples/hypervisor_manager.tf (hunk 12 above)
Comment (nabhajit-ray) [reply examples/hypervisor_manager.tf:1]: Done

**Pull request merged**